chrome/content/zotero/zoteroPane.js

```diff
@@ -4491,7 +4491,8 @@ var ZoteroPane = new function()
 	
 	
 	this.addAttachmentFromURI = Zotero.Promise.method(function (link, itemID) {
-		if (!this.canEdit()) {
+		if (Zotero_Tabs.selectedID === 'zotero-pane' && !this.canEdit()
+				|| !Zotero.Items.get(itemID).library.editable) {
```

Comment:
It seems kind of confusing to leave `canEdit()` and `canEditFiles()` checking just the tree but have other logic in this same file check for other situations. Shouldn't we just have those functions check editability based on the current context?